# Introduction To Probability
## Challenge 1

A and B are events of a probability space with $(\omega, \sigma, P)$ such that $P(A) = 0.3$, $P(B) = 0.6$ and $P(A \cap B) = 0.1$

Which of the following statements are false?
* $P(A \cup B) = 0.6$
* $P(A \cap B^{C}) = 0.2$
* $P(A \cap (B \cup B^{C})) = 0.4$
* $P(A^{C} \cap B^{C}) = 0.3$
* $P((A \cap B)^{C}) = 0.9$

In [9]:
from sympy import *

# Definir los símbolos y eventos
A, B = symbols('A B')
PA = 0.3
PB = 0.6
P_interseccion = 0.1

# Definir las probabilidades de los eventos
P_A = Rational(PA)
P_B = Rational(PB)
P_A_interseccion_B = Rational(P_interseccion)

# Calcular la probabilidad de la unión de A y B
P_union = P_A + P_B - P_A_interseccion_B
print("P(A ∪ B) =", float(P_union))

# Calcular la probabilidad de la intersección de A y no B
P_A_interseccion_noB = P_A - P_A_interseccion_B
print("P(A ∩ B') =", float(P_A_interseccion_noB))

# Calcular la probabilidad de A intersección (B unión B')
P_B_union_noB = P_B + (1 - P_B)
P_A_interseccion_B_union_noB = P_A * P_B_union_noB
print("P(A ∩ (B ∪ B')) =", float(P_A_interseccion_B_union_noB))

# Calcular la probabilidad de la no A intersección no B
P_noA = 1 - P_A
P_noB = 1 - P_B
P_noA_interseccion_noB = P_noA * P_noB
print("P(A' ∩ B') =", float(P_noA_interseccion_noB))

# Calcular la probabilidad de la intersección de no (A y B)
P_noA_interseccion_noB = 1 - P_A_interseccion_B
print("P((A ∩ B)') =", float(P_noA_interseccion_noB))


P(A ∪ B) = 0.7999999999999999
P(A ∩ B') = 0.19999999999999998
P(A ∩ (B ∪ B')) = 0.3
P(A' ∩ B') = 0.28
P((A ∩ B)') = 0.9


In [ ]:
𝑃(𝐴∪𝐵)=0.6        ### que ocurra A o B
 
𝑃(𝐴∩𝐵𝐶)=0.2       ### 
 
𝑃(𝐴∩(𝐵∪𝐵𝐶))=0.4   ###   falso
 
𝑃(𝐴𝐶∩𝐵𝐶)=0.3       ###   falso
 
𝑃((𝐴∩𝐵)𝐶)=0.9      ### verdadero (𝑃(𝐴∩𝐵)=0.1) + (𝑃((𝐴∩𝐵)𝐶)=0.9 ) = 1

## Challenge 2
There is a box with 10 white balls, 12 red balls and 8 black balls. Calculate the probability of:
* Taking a white ball out.
* Taking a white ball out after taking a black ball out.
* Taking a red ball out after taking a black and a red ball out.
* Taking a red ball out after taking a black and a red ball out with reposition.

**Hint**: Reposition means putting back the ball into the box after taking it out.

In [1]:

blancas = 10/30 *100
print(blancas)
blancas_negra = (8/30)* (1/3)
print(blancas_negra)
red_blackred = (20/30 * 19/29) * (12/28)
print(red_blackred)
red_blackred2 = (20/30 * 20/30) * (12/30)
print(red_blackred2)

33.33333333333333
0.08888888888888888
0.18719211822660095
0.17777777777777778


33%,
8%,
18%,
17%

## Challenge 3

You are planning to go on a picnic today but the morning is cloudy. You hate rain so you don't know whether to go out or stay home! To help you make a decision, you gather the following data about rainy days:

* Knowing that it is a rainy day, the probability of cloudy is 50%!
* The probability of any day (rainy or not) starting off cloudy is 40%. 
* This month is usually dry so only 3 of 30 days (10%) tend to be rainy. 

What is the probability of rain, given the day started cloudy?

In [2]:
0.5 * 0.1 / 0.4

0.125

12.5%

## Challenge 4

One thousand people were asked through a telephone survey whether they thought more street lighting is needed at night or not.

Out of the 480 men that answered the survey, 324 said yes and 156 said no. On the other hand, out of the 520 women that answered, 351 said yes and 169 said no. 

We wonder if men and women have a different opinions about the street lighting matter. Is gender relevant or irrelevant to the question?

Consider the following events:
- The answer is yes, so the person that answered thinks that more street lighting is needed.
- The person who answered is a man.

We want to know if these events are independent, that is, if the fact of wanting more light depends on whether one is male or female. Are these events independent or not?

**Hint**: To clearly compare the answers by gender, it is best to place the data in a table.

In [5]:
import pandas as pd

In [7]:
dicc = {"men": [324, 156],
       "women": [351, 169],
       "total": [675, 325]}

In [8]:
dicc = pd.DataFrame(dicc)

In [9]:
dicc

men  women  total
0  324    351    675
1  156    169    325

In [10]:
P_man_yes = 324/480

In [11]:
P_man_yes

0.675

In [14]:
P_women_yes = 351/520

In [15]:
P_women_yes

0.675

eventos son independientes.